tests/unit/general/test_umpire_mem.cpp

```diff
@@ -123,10 +127,18 @@ static void test_umpire_device_memory()
    printf("perm=%ld, temp=%ld\n", alloc_size(permanent), alloc_size(temporary));
 
    // pinned host memory
-   Vector pinned_host_perm(num_elems, mfem::MemoryType::HOST_PINNED);
+   auto orig_host_mt = Device::GetHostMemoryType();
+   Device::SetHostMemoryType(MemoryType::HOST_PINNED);
+   Vector pinned_host_perm(num_elems);
+   // alternative, without switching the global host MemoryType:
+   // Vector pinned_host_perm(num_elems, MemoryType::HOST_PINNED);
```

Comment:
This (the commented out code) is what we'd do in practice. The current set/restore MT approach is actually what we started with but we didn't really like it because it isn't always clear when allocations are happening when someone is using higher level mfem interfaces.

Reply:
That is a good point, in general.

In this case, the host allocation happens right away, so this should work.

However, as you said, the actual moment of allocation in other cases may be more difficult to track.